src/client/schedules/types.ts

```diff
@@ -3,349 +3,271 @@
-	 */
-	nextChange: HDateTime
-
-	/**
-	 * Is the list of calendar(s) the schedule subscribes to
-	 */
-	calendars?: HList<HRef>
+export interface WeekDaySchedule {
+	dayOfWeek: HNum
+	dailySchedule: DailySchedule[]
+}
 
-	/**
-	 * Specifies the list of enumerated values that the schedule value can have. Only applicable
-	 * when the schedule is of the type string
-	 */
-	enum?: HList<HStr>
+/**
+ * A List of Schedules Exceptions.
+ */
+export type ExceptionSchedule = SpecialEvent[]
 
-	/**
-	 * Specifies the default value of the schedule (if any)
-	 */
-	defaultVal?: HVal
+/**
+ * Defines data for a special event.
+ */
+export interface SpecialEvent {
+	calendarEntry?: CalendarEntry
+	calendarRef?: HRef
+	priority: number
+	dailySchedule: DailySchedule[]
+}
 
-	/**
-	 * Is the range of dates (if any) that the schedule is active
-	 */
-	range?: ScheduleRange
+/**
+ * Defines the schedule for a day.
+ */
+export interface DailySchedule {
+	bacnetTime: {
+		hour: number
+		minute: number
+		second: number
+		hundredths: number
+	}
+	scheduledVal: HVal
 }
 
 /**
- * The Calendar object
+ * The Calendar object.
  */
 export interface Calendar extends HDict {
 	/**
-	 * Specifies the id of the calendar
+	 * Is the id of the calendar object.
 	 */
-	id: HRef
+	id?: HRef
 
 	/**
-	 * Is the date-time when the calendar was last modified
+	 * Specifies the name of the calendar.
 	 */
-	mod: HDateTime
+	dis?: HStr
 
 	/**
-	 * Specifies the display name/label of the calendar
+	 * Is the date-time when the calendar was last modified
 	 */
-	dis: HStr
+	mod?: HDateTime
 
 	/**
-	 * Specifies the value of the calendar object (if any)
+	 * The Marker tag describing as calendar.
 	 */
-	val?: HStr
+	calender: HMarker
 
 	/**
-	 * Specifies the parent calendar object (if any)
+	 * The Ref of the calendar.
 	 */
-	extends?: HRef
+	calendarRef?: HRef
 
 	/**
-	 * Specifies the list of all the entries that the calendar is active
+	 * The calendar entries
 	 */
-	entries: HList<CalendarEntry>
+	entries?: CalendarEntry[]
 }
 
 /**
- * Object defining the schedule point
+ * A Calendar Entry.
  */
-export interface SchedulePoint extends HDict {
-	/**
-	 * Is the id of the point
-	 */
-	id: HRef
+export type CalendarEntry =
+	| ({ entryType: 'Date' } & CalendarDate)
+	| ({ entryType: 'Range' } & CalendarRange)
+	| ({ entryType: 'WeekNDay' } & CalendarWeekNDay)
 
-	/**
-	 * Is the number specifying to what priority on the point should the
-	 * schedule be writing on. Range 1-15
-	 */
-	schedulable: HNum
+/**
+ * A Date for a Calendar.
+ */
+export interface CalendarDate {
+	year: number
+	month: number
+	dayOfMonth: number
+	dayOfWeek: number
+}
 
-	/**
-	 * Is the schedule that is writing to the point
-	 */
-	scheduleRef: HRef
+/**
+ * The Date range of a Calendar.
+ */
+export interface CalendarRange {
+	lowBound?: HDate
+	upBound?: HDate
+}
+
+/**
+ * Define the week and day of a Calendar.
+ */
+export interface CalendarWeekNDay {
+	nthWeek: number
+	dayOfWeek: number
```

Comment:
`month` is missing. `CalendarWeekNDay` has three components: `dayOfWeek`, `weekOfMonth` and `month`.